violinplot.m

```diff
@@ -44,11 +44,25 @@
 %                    Defaults to false
 %     'ShowMean'     Whether to show mean indicator
 %                    Defaults to false
+%     'GroupOrder'   Cell of category names in order to be plotted.
+%                    Defaults to alphabetical ordering
 
 % Copyright (c) 2016, Bastian Bechtold
 % This code is released under the terms of the BSD 3-clause license
 
     hascategories = exist('cats','var') && not(isempty(cats));
+    grouporder = {};
+    ii=1;
+    while ii<=numel(varargin)
```

Comment:
You can replace most of this loop with something like `find(strcmp(varargin, 'GroupOrder'))`. This will clean up this part of the code considerably.

Reply:
Done, added some more error checking too